routesrv/routesrv.go

```diff
@@ -79,13 +81,60 @@ func New(opts Options) (*RouteServer, error) {
 	if err != nil {
 		return nil, err
 	}
+
+	if opts.SecretsRegistry == nil {
+		opts.SecretsRegistry = secrets.NewRegistry()
+	}
+	defer opts.SecretsRegistry.Close()
+
+	sp := secrets.NewSecretPaths(opts.CredentialsUpdateInterval)
+	defer sp.Close()
+	for _, p := range opts.CredentialsPaths {
+		if err := sp.Add(p); err != nil {
+			log.Errorf("Failed to add credentials file: %s: %v", p, err)
+		}
+	}
+
+	oauthConfig := &auth.OAuthConfig{}
+	if opts.EnableOAuth2GrantFlow /* explicitly enable grant flow */ {
+		grantSecrets := secrets.NewSecretPaths(opts.CredentialsUpdateInterval)
+		defer grantSecrets.Close()
+
+		oauthConfig.AuthURL = opts.OAuth2AuthURL
+		oauthConfig.TokenURL = opts.OAuth2TokenURL
+		oauthConfig.RevokeTokenURL = opts.OAuth2RevokeTokenURL
+		oauthConfig.TokeninfoURL = opts.OAuthTokeninfoURL
+		oauthConfig.SecretFile = opts.OAuth2SecretFile
+		oauthConfig.ClientID = opts.OAuth2ClientID
+		oauthConfig.ClientSecret = opts.OAuth2ClientSecret
+		oauthConfig.ClientIDFile = opts.OAuth2ClientIDFile
+		oauthConfig.ClientSecretFile = opts.OAuth2ClientSecretFile
+		oauthConfig.CallbackPath = opts.OAuth2CallbackPath
+		oauthConfig.AuthURLParameters = opts.OAuth2AuthURLParameters
+		oauthConfig.SecretsProvider = grantSecrets
+		oauthConfig.Secrets = opts.SecretsRegistry
+		oauthConfig.AccessTokenHeaderName = opts.OAuth2AccessTokenHeaderName
+		oauthConfig.TokeninfoSubjectKey = opts.OAuth2TokeninfoSubjectKey
+		oauthConfig.TokenCookieName = opts.OAuth2TokenCookieName
+		oauthConfig.ConnectionTimeout = opts.OAuthTokeninfoTimeout
+		oauthConfig.Tracer = tracer
```

Comment:
Do we need all of this?
I think grant pre-processor only needs `OAuth2CallbackPath`

Reply:
yeah we don't need all of this, I thought we need to call `Init` which is not true here.